queue/src/main/java/org/killbill/bus/DefaultPersistentBus.java

```diff
@@ -138,39 +149,44 @@ public void start() {
             return;
         }
 
-        if (isStarted.compareAndSet(false, true)) {
+        if (isInitialized.compareAndSet(false, true)) {
             dao.initialize();
             dispatcher.start();
             startQueue();
-        }
 
-        if (config.getPersistentQueueMode() == PersistentQueueMode.STICKY_POLLING || config.getPersistentQueueMode() == PersistentQueueMode.STICKY_EVENTS) {
-            reaper.start();
+            if (config.getPersistentQueueMode() == PersistentQueueMode.STICKY_POLLING || config.getPersistentQueueMode() == PersistentQueueMode.STICKY_EVENTS) {
+                reaper.start();
+            }
+            isStarted.set(true);
```

Comment:
I am tempted to say 'yes' (with the way our start() method have been written) : The `isInitialized` is needed to ensure only one thread runs the `start()` method; the `isStarted` is needed to ensure nothing can happen prior completed the `start() ` method.

If i do...

```
        if (isStarted.compareAndSet(false, true)) {
            dao.initialize();
            dispatcher.start();
            startQueue();

            if (config.getPersistentQueueMode() == PersistentQueueMode.STICKY_POLLING || config.getPersistentQueueMode() == PersistentQueueMode.STICKY_EVENTS) {
                reaper.start();
            }
        }
```

... then, only one thread will run the block, but the `isStarted` flag is set right away to true prior the code completed: Since this code now potentially pulls entries from the DB (ini sequence), it's important to have other threads wait for the init to have completed.